lua/neoclip/sorted_set.lua

```diff
@@ -47,6 +47,21 @@ local insert = function(self, entry, opts)
     end
 end
 
+local replace = function(self, entry, newEntry)
+    local key = hash(entry)
+    local new_key = hash(newEntry)
+    local node = self.entries[key];
+    node.value = newEntry -- Replace entry in node.
+
+    -- Prevent duplicate entries
+    if self.entries[new_key] and key ~= new_key then
+        self:remove(newEntry)
+    end
```

Comment:
While implementing the new tests, I realized my implementation allowed for duplicate entries to exist in the linked list. This is a problem because they will both have the same hash, and the dictionary won't be able to track them both. These three lines fix that

Reply:
Cool!